k9mail-library/src/main/java/com/fsck/k9/mail/internet/MimeMessage.java

```diff
@@ -317,14 +317,40 @@ public void setSubject(String subject) {
     public Address[] getFrom() {
         if (mFrom == null) {
             String list = MimeUtility.unfold(getFirstHeader("From"));
-            if (list == null || list.length() == 0) {
+            if (list == null || list.isEmpty()) {
                 list = MimeUtility.unfold(getFirstHeader("Sender"));
             }
+            if (list == null || list.isEmpty()) {
```

Comment:
do we need to check the same condition twice? The line below can be in the IF above, right?

Reply:
The thing is that list can still be null or empty after calling the MimeUtility.unfold. So we need the check. If it does not look good then maybe some function can be extracted.

Reply:
right...silly me, still half sleeping this morning, of course, forget the comment 👍

Reply:
Please check now, I improved the logic to avoid repetition.

Reply:
very nice, so also future headers check are also possible by adding to the list. Cool